src/pages/trainer/training/edit/components/toolbar/buttons/toolbarButton.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+interface IProps {
+  editor: HTMLTextAreaElement;
+  caret: string;
+  offset: number;
+  onClick: (editor: HTMLTextAreaElement, caret: string, offset: number) => void;
```

Comment:
Here we need some discussion, maybe this approach is fine, but IMHO i would just send the onClick to the parent component indicating the selected command, then the parent component takes care of doing the right plumbing, what do you think?